src/Raven.Server/Documents/Handlers/Processors/Replication/AbstractPullReplicationHandlerProcessorForRegisterHubAccess.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client.Documents.Operations.Replication;
+using Raven.Client.Exceptions;
+using Raven.Client.Json.Serialization;
+using Raven.Client.Util;
+using Raven.Server.Documents.Handlers.Processors.Databases;
+using Raven.Server.ServerWide.Commands;
+using Raven.Server.ServerWide.Context;
+using Raven.Server.Web;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Handlers.Processors.Replication
+{
+    internal abstract class AbstractPullReplicationHandlerProcessorForRegisterHubAccess<TRequestHandler> : AbstractHandlerProcessorForUpdateDatabaseConfiguration<BlittableJsonReaderObject, TRequestHandler>
+        where TRequestHandler : RequestHandler
+    {
+        private string _hubTaskName;
+        private PullReplicationDefinition _hubDefinition;
+
+        protected AbstractPullReplicationHandlerProcessorForRegisterHubAccess([NotNull] TRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask<BlittableJsonReaderObject> GetConfigurationAsync(TransactionOperationContext context, string databaseName, AsyncBlittableJsonTextWriter writer)
+        {
+            _hubTaskName = RequestHandler.GetStringQueryString("name", true);
+            using (context.OpenReadTransaction())
```

Comment:
let's avoid keeping the read tx open while we are reading from the network

Reply:
done